binaries/recoverytest/main.go

```diff
@@ -0,0 +1,92 @@
+package main
+
+import (
+	"log"
+	"sync"
+	"time"
+
+	"github.com/scootdev/scoot/tests/testhelpers"
+)
+
+func main() {
+
+	// RecoverTest Parameters
+	numJobs := 20
+	timeout := 20 * time.Second
+
+	var wg sync.WaitGroup
+	scootClient := testhelpers.CreateScootClient("localhost:9090")
+
+	// Initialize Local Cluster
+	cluster1Cmds, err := testhelpers.CreateLocalTestCluster()
+	if err != nil {
+		log.Fatalf("Unexpected Error while Setting up Local Cluster %v", err)
+	}
+	defer cluster1Cmds.Kill()
+
+	testhelpers.WaitForClusterToBeReady(scootClient)
+
+	// Add a Bunch of Jobs to Scoot CloudExec
+	log.Printf("Add Jobs to Scoot Cloud Exec")
+	jobIds := make([]string, 0, numJobs)
+	for i := 0; i < numJobs; i++ {
+		id, err := testhelpers.GenerateAndStartJob(scootClient)
+		if err != nil {
+			log.Fatalf("Could not schedule Jobs Error: %v", err)
+		} else {
+			jobIds = append(jobIds, id)
+		}
+	}
+
+	// Wait for jobs to complete check in a separate go routine
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		err := testhelpers.WaitForJobsToCompleteAndLogStatus(jobIds, scootClient, timeout)
+		if err != nil {
+			log.Fatalf("Error Occurred Waiting For Jobs to Complete.  %v", err)
+		}
+	}()
+
+	// Wait for Jobs to Start Running and make some progress
+	time.Sleep(1 * time.Second)
```

Comment:
So it could in theory complete before killing the cluster, but in practice won't with 20 tests that just sleep for 500ms.  

We are guaranteed because I wait till the jobs are started, that they are all in the scheduler queue, if there was some really bad lag they might not start but that would be really weird on a brand new clean cluster.  We also have a unit test already that verifies that the 500ms jobs run in a reasonable amount of time, ensures we are not leaving idle workers with scheduling etc...

We could improve this in the future by doing a check that a percentage of tasks are in progress or completed before killing the cluster in the future as we build out the test harness.